src/scripts/main.js

```diff
@@ -1,3 +1,25 @@
 'use strict';
 
-// write code here
+const transformToNumber = (str) => {
+  return parseFloat(str.replace(/[$,]/g, ''));
+};
+const getEmployees = (list) => {
+  const result = list.map((employee) => {
+    return {
+      name: employee.textContent.trim(),
+      salary: transformToNumber(employee.getAttribute('data-salary')),
+    };
+  });
+
+  return SortList(result);
+};
+const SortList = (list) => {
+  return list.sort((a, b) => b.salary - a.salary);
+};
+
+const employeesArray = Array.from(document.querySelectorAll('ul li'));
+const employeesData = employeesArray.map((employee) => {
+  getEmployees(employee);
+});
+
+SortList(employeesData);
```

Comment:
This line is redundant because `getEmployees` already sorts the list. You can remove this line to avoid unnecessary sorting.